src/Pimple/Configurator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace TomPHP\ConfigServiceProvider\Pimple;
+
+use Pimple\Container;
+use ReflectionClass;
+use TomPHP\ConfigServiceProvider\ApplicationConfig;
+use TomPHP\ConfigServiceProvider\ContainerConfigurator;
+use TomPHP\ConfigServiceProvider\InflectorConfig;
+use TomPHP\ConfigServiceProvider\ServiceConfig;
+
+final class Configurator implements ContainerConfigurator
+{
+    public function addApplicationConfig($container, ApplicationConfig $config, $prefix = 'config')
+    {
+        if (!empty($prefix)) {
+            $prefix .= $config->getSeparator();
+        }
+
+        foreach ($config as $key => $value) {
+            $container[$prefix . $key] = $value;
+        }
+    }
+
+    public function addServiceConfig($container, ServiceConfig $config)
+    {
+        foreach ($config as $definition) {
+            $factory = function () use ($definition) {
+                $reflection = new ReflectionClass($definition->getClass());
+
+                $instance = $reflection->newInstanceArgs($definition->getArguments());
```

Comment:
If minimum support was for `5.6` then all of this could be made much nicer with varidic arguments.